### DISCUSSION TITLE
[2021/12/07] Frontend Night

### DISCUSSION THREAD

**Discussion by jjenko** [wantedly/frontend-night#182] [Archive]
## Policy

- 資料の準備をがんばらない
- 参加・途中退席は自由
- スキップしない

話したい・聞きたいネタを書いてく ✏️

ハッシュタグ [#frontend_night](https://twitter.com/hashtag/frontend_night?src=hash)

Prev: https://github.com/wantedly/frontend-night/discussions/181 

 _Created from https://github.com/wantedly/frontend-night/discussions/179 by [issue-creator](https://github.com/rerost/issue-creator)_

**Comment by hayaokimura** [comment]
https://developers.freee.co.jp/entry/a11y-training
アクセシビリティをどれぐらい気にするか悩ましいけどこの資料は良い資料

**Comment by NotFounds** [comment]
## JavaScript の Date module で踏みやすいバグ
https://github.com/denoland/deno_std/issues/1491

- Deno の標準ライブラリとして deno_std があり、日付操作用の便利関数として [datetime module](https://deno.land/std@0.117.0/datetime) がある。今回はその中にある parse 関数のできごと
- datetime/parse はフォーマットされた日付文字列から JavaScript の `Date` オブジェクトに変換する
- 2021/10/31 に冒頭の issue が立てられた
  - 大きい月(31まである月)に parse を使うと意図しない結果となってしまうというバグ報告
- 原因は parse の中で使っている `Date` module の `Date.prototype.setMonth` の仕様の問題
  - 正しく使うには `setMonth(month, date)` のように date を指定してあげることが必要

**`Date` module はやっぱり難しいので生の API は触りたくないしなるべく触らないほうが安全**
